packages/anasazi/tpetra/test/IRTR/cxx_main.cpp

```diff
@@ -0,0 +1,253 @@
+// @HEADER
+// *****************************************************************************
+//                 Anasazi: Block Eigensolvers Package
+//
+// Copyright 2004 NTESS and the Anasazi contributors.
+// SPDX-License-Identifier: BSD-3-Clause
+// *****************************************************************************
+// @HEADER
+//
+// This test is for IRTR solving a generalized (Ax=Bxl) real Hermitian
+// eigenvalue problem, using the RTRSolMgr solver manager.
+//
+#include "AnasaziConfigDefs.hpp"
+#include "AnasaziTypes.hpp"
+
+#include "AnasaziTpetraAdapter.hpp"
+#include "AnasaziBasicEigenproblem.hpp"
+#include "AnasaziRTRSolMgr.hpp"
+#include <Teuchos_CommandLineProcessor.hpp>
+#include <Teuchos_StandardCatchMacros.hpp>
+#include <Teuchos_SerialDenseMatrix.hpp>
+
+#include <Tpetra_Core.hpp>
+#include <Tpetra_CrsMatrix.hpp>
+
+#include <ModeLaplace1DQ1.hpp>
+
+using namespace Teuchos;
+
+int main(int argc, char *argv[]) 
+{
+
+  using std::cout;
+  using std::endl;
+
+  bool boolret;
+
+  Tpetra::ScopeGuard tpetraScope (&argc,&argv);
+  RCP<const Teuchos::Comm<int> > comm = Tpetra::getDefaultComm ();
+  int MyPID = comm->getRank();
+
+  bool testFailed = false;
+  bool verbose = false;
+  bool debug = false;
+  bool shortrun = false;
+  bool skinny = true;
+  bool useSA = false;
+  std::string which("SR");
+  int numElements = 100;
+  int user_verbosity = 0;
+  int numicgs = 2;
+
+  bool success = true;
+  try {
+
+    CommandLineProcessor cmdp(false,true);
+    cmdp.setOption("verbose","quiet",&verbose,"Print messages and results.");
+    cmdp.setOption("debug","nodebug",&debug,"Print debugging information.");
+    cmdp.setOption("sort",&which,"Targetted eigenvalues (SR or LR).");
+    cmdp.setOption("shortrun","longrun",&shortrun,"Allow only a small number of iterations.");
+    cmdp.setOption("skinny","hefty",&skinny,"Use a skinny (low-mem) or hefty (higher-mem) implementation of IRTR.");
+    cmdp.setOption("numElements",&numElements,"Number of elements in discretization.");
+    cmdp.setOption("useSA","noSA",&useSA,"Use subspace acceleration.");
+    cmdp.setOption("verbosity",&user_verbosity,"Additional verbosity falgs.");
+    cmdp.setOption("numICGS",&numicgs,"Num ICGS iterations");
+    if (cmdp.parse(argc,argv) != CommandLineProcessor::PARSE_SUCCESSFUL) {
+      return -1;
+    }
+    if (debug) verbose = true;
+
+    typedef double                              ST;
```

Comment:
Would it make sense to test not only for double, but any scalar type that's enabled in Tpetra?

Reply:
Test is only meaningful for double-precision due to generalized eigenvalue problem being generated.